tests/print.cpp

```diff
@@ -0,0 +1,268 @@
+// Copyright (C) The DDC development team, see COPYRIGHT.md file
+//
+// SPDX-License-Identifier: MIT
+
+#include <fstream>
+#include <iomanip>
+#include <sstream>
+#include <string>
+
+#include <ddc/ddc.hpp>
+
+#include <gtest/gtest.h>
+
+#include <Kokkos_Core.hpp>
+#include <Kokkos_Random.hpp>
+
+inline namespace anonymous_namespace_workaround_print_cpp {
+
+struct DDim0
+{
+};
+struct DDim1
+{
+};
+struct DDim2
+{
+};
+struct DDim3
+{
+};
+struct DDim4
+{
+};
+struct DDim5
+{
+};
+
+} // namespace anonymous_namespace_workaround_print_cpp
+
+template <typename cell>
+void PrintTestCheckOutput0d()
+{
+    ddc::DiscreteDomain<> const domain_full;
+
+    ddc::Chunk cells_in_dev_alloc("cells_in_dev", domain_full, ddc::DeviceAllocator<cell>());
+    ddc::ChunkSpan const cells_in = cells_in_dev_alloc.span_view();
+
+    ddc::parallel_for_each(
+            domain_full,
+            KOKKOS_LAMBDA(ddc::DiscreteElement<> const i) { cells_in(i) = 0.12345; });
+
+    {
+        std::stringstream ss;
+        print_content(ss, cells_in);
+        EXPECT_EQ("0.12345", ss.str());
+    }
+}
+
+TEST(Print, CheckOutput0d)
+{
+    PrintTestCheckOutput0d<float>();
+    PrintTestCheckOutput0d<double>();
+}
+
+template <typename cell>
+void TestPrintCheckOutput2d()
+{
+    unsigned const dim0 = 2;
+    unsigned const dim1 = 2;
+
+    ddc::DiscreteDomain<DDim0> const domain_0
+            = ddc::init_trivial_bounded_space(ddc::DiscreteVector<DDim0>(dim0));
+    ddc::DiscreteDomain<DDim1> const domain_1
+            = ddc::init_trivial_bounded_space(ddc::DiscreteVector<DDim1>(dim1));
+
+    ddc::DiscreteDomain<DDim0, DDim1> const domain_2d(domain_0, domain_1);
+
+    ddc::Chunk cells_in_dev_alloc("cells_in_dev", domain_2d, ddc::DeviceAllocator<cell>());
+    ddc::ChunkSpan const cells_in = cells_in_dev_alloc.span_view();
+
+    ddc::parallel_for_each(
+            domain_2d,
+            KOKKOS_LAMBDA(ddc::DiscreteElement<DDim0, DDim1> const i) { cells_in(i) = 0.12345; });
+
+    cells_in(ddc::DiscreteElement<DDim0, DDim1>(1, 1)) = -0.12345;
```

Comment:
```suggestion
    cells_in(domain_2d.front() + ddc::DiscreteVector<DDim0, DDim1>(1, 1)) = -0.12345;
```

Note that it will fail on a GPU build.

Reply:
This was removed, now the values are set together in the parallel_for